packages/sdk/src/auth/token-provider.ts

```diff
@@ -59,6 +61,7 @@ export class CachedTokenProvider extends TokenProvider {
   constructor(
     private readonly delegate: TokenProvider,
     private readonly tokenStore: TokenStore,
+    private readonly implicitWalletCreation: boolean,
```

Comment:
Can I suggest to change this from `boolean` to `type WalletCreation = 'default' | 'implicit'`?

Reply:
But in dialect-cloud config have `implicitWalletCreation `?

Reply:
For now `default`='implicit'. What did you mean when you sugges 'default'?

Reply:
I assume @fsher meant to have it as `{ walletCreation: WalletCreation }`, but in this case it make sense to make the `default` = implicit upserting, and `explicit` as another option